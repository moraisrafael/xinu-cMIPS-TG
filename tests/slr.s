	.include "cMIPS.s"
	.text
	.align 2
	.set noat
	.globl _start
	.ent _start
_start: nop
	la $15,x_IO_BASE_ADDR
	la   $3,0x80000000
	nop
	nop
	nop
	srl $4,$3,1
	sw  $4, 4($15)  	# 0x8000.0000 -> 0x4000.0000
	srl $4,$3,2
	sw  $4, 0($15)		# 0x2000.0000
	srl $4,$3,3
	sw  $4, 4($15)		# 0x1000.0000
	srl $4,$3,4
	sw  $4, 0($15)		# 0x0800.0000
	srl $4,$3,5
	sw  $4, 4($15)		# 0x0400.0000
	srl $4,$3,6
	sw  $4, 0($15)		# 0x0200.0000
	srl $4,$3,30
	sw  $4, 4($15)		# 2
	srl $4,$3,31
	sw  $4, 0($15)		# 1
	nop
	sw  $0, 0($15)
	sw  $0, 0($15)
	nop
	la  $15,x_IO_BASE_ADDR
	la  $8,0x01010101
	sw  $8, 4($15)    # 16843009 = 0101.0101
	srl $4,$8,1
	sw  $4, 0($15)    # 8421504 = 0080.8080
	srl $4,$8,2
	sw  $4, 4($15)    # 4210752 = 00404040
	srl $4,$8,5
	sw  $4, 0($15)   # 526344 = 0008.0808
	srl $4,$8,16
	sw  $4, 4($15)	  # 257 = 0000.0101
	srl $4,$8,31
	sw  $4, 0($15)   # 0
	sw  $0, 0($15)
	sw  $0, 0($15)
	li   $1,31
incr:	li   $8, 0x80000000
	srlv $4,$8,$1
	addi $1,$1,-4
	sw   $4, 4($15)   # 1,16,256,4096,65536,1048576,16777216,268435456
	slt  $22,$0,$1    # 0x0000.0001,0x0000.0010,0x0000.0100,0x0000.1000,
	bne  $22,$0,incr  # 0x0001.0000,0x0010.0000,0x0100.0000,0x1000.0000
	sw  $0, 0($15)
	sw  $0, 0($15)
	li   $1,31
arith:	li   $8, 0x80000000
	srav $4,$8,$1
	addi $1,$1,-4
	sw   $4, 4($15)   # 1,16,256,4096,65536,1048576,16777216,268435456
	slt  $22,$0,$1    # 0xFFFF.FFFF,0xFFFF.FFF0,0xFFFF.FF00,0xFFFF.F000,
	bne  $22,$0,arith # 0xFFFF.0000,0xFFF0.0000,0xFF00.0000,0xF000.0000
	nop
	wait
	.end _start

	