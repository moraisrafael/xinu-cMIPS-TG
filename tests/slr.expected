40000000
20000000
10000000
08000000
04000000
02000000
00000002
00000001
00000000
00000000
01010101
00808080
00404040
00080808
00000101
00000000
00000000
00000000
00000001
00000010
00000100
00001000
00010000
00100000
01000000
10000000
00000000
00000000
ffffffff
fffffff0
ffffff00
fffff000
ffff0000
fff00000
ff000000
f0000000
